src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -620,12 +620,13 @@ private KernelUpdateDeploymentTask createKernelUpdateDeployment(Deployment deplo
         return new KernelUpdateDeploymentTask(kernel, logger.createChild(), deployment, componentManager);
     }
 
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
     private DefaultDeploymentTask createDefaultNewDeployment(Deployment deployment) {
         try {
             logger.atInfo().kv("document", deployment.getDeploymentDocument())
                     .log("Received deployment document in queue");
             parseAndValidateJobDocument(deployment);
-        } catch (InvalidRequestException e) {
+        } catch (Exception e) {
```

Comment:
Add it to a different catch block as the messaging is not appropriate. Keep as much specific exception as we can. 
What exception are you seeing btw?

Reply:
we are seeing null pointer exception because version field for a root component in the deployment document can be null.

Reply:
This is probably fine if you capture the specific exception before this.